File: domain/src/main/java/org/hesperides/domain/workshopproperties/entities/WorkshopProperty.java
package org.hesperides.domain.workshopproperties.entities;

import lombok.Value;

@Value
public class WorkshopProperty {
    String key;
    String value;
    String keyValue;

    public WorkshopProperty concatKeyValue() {
        return new WorkshopProperty(
                this.key,
                this.value,
                this.key+this.value
        );
    }
}

File: domain/src/main/kotlin/org/hesperides/domain/workshopproperties/WorkshopProperty.kt
package org.hesperides.domain.workshopproperties

import org.axonframework.commandhandling.TargetAggregateIdentifier
import org.hesperides.domain.security.User
import org.hesperides.domain.security.UserEvent
import org.hesperides.domain.workshopproperties.entities.WorkshopProperty

// Commands
data class CreateWorkshopPropertyCommand(val workshopProperty: WorkshopProperty, val user: User)

data class UpdateWorkshopPropertyCommand(@TargetAggregateIdentifier val key: String,
                                         val workshopProperty: WorkshopProperty,
                                         val user:User)

// Events
data class WorkshopPropertyCreatedEvent(val workshopProperty: WorkshopProperty,
                                        override val user: User) : UserEvent(user)

data class WorkshopPropertyUpdatedEvent(val workshopProperty: WorkshopProperty,
                                        override val user: User) : UserEvent(user)

// Queries
data class GetWorkshopPropertyByKeyQuery(val platformKey: String)